server.go

```diff
@@ -297,78 +294,34 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 
 		// configure Datadog as sink
 		if ret.DDTraceAddress != "" {
-			ret.tracerSinks = append(ret.tracerSinks, tracerSink{
-				name:   "Datadog",
-				tracer: nil,
-				flush:  flushSpansDatadog,
-			})
-			log.Info("Configured Datadog trace sink")
-		}
 
-		// configure Lightstep as Sink
-		if ret.traceLightstepAccessToken != "" {
-			var host *url.URL
-			host, err = url.Parse(conf.TraceLightstepCollectorHost)
+			var ddSink spanSink
```

Comment:
Just because the function uses named returns, that doesn't mean you're forced to use them everywhere. Since the non-error variable is intended to be local, I'd use `:=` in assignment and then return `err` explicitly in the nil check. (When I see non-`:=` assignment for return values, I assume that the non-error value is intended to be outer-scoped, which isn't the case here)


I'd actually make the broader argument that this function shouldn't be using named returns at all, but that's a yak to shave another day.